common/main/java/com/couchbase/lite/LiveQuery.java

```diff
@@ -192,38 +196,34 @@ private void refreshResults() {
                 prevResults = previousResults;
             }
 
-            final ResultSet newResults;
-            if (prevResults == null) { newResults = query.execute(); }
-            else { newResults = prevResults.refresh(); }
-            Log.i(DOMAIN, "LiveQuery refresh: %s > %s", prevResults, newResults);
+            // both of these methods seize the db lock
+            final ResultSet newResults = (prevResults == null) ? query.execute() : prevResults.refresh();
+            Log.i(DOMAIN, "LiveQuery refresh: %s -> %s", prevResults, newResults);
 
             if (newResults == null) { return; }
 
-            // ??? This could close a result set to which
-            // client code still has a reference.
-            if (prevResults != null) { prevResults.forceClose(); }
+            if (prevResults != null) { prevResults.close(); }
 
             boolean update = false;
             synchronized (lock) {
                 if (state.get() != State.STOPPED) {
-                    newResults.retain();
```

Comment:
I reason to myself that this can safely be removed because it will stay alive as long as the LiveQuery Java object.  Is that correct?  Otherwise I worry about below calling `newResults.copy()` at an arbitrary time on a potentially dead object.